mlflow/utils/search_utils.py

```diff
@@ -104,14 +86,12 @@ def _get_identifier(cls, identifier):
         try:
             entity_type, key = identifier.split(".", 1)
         except ValueError:
-            raise MlflowException("Invalid filter string '%s'. Filter comparison is expected as "
-                                  "'attribute.<key> <comparator> <value>', "
-                                  "'metric.<key> <comparator> <value>', "
-                                  "'tag.<key> <comparator> <value>', or "
-                                  "'params.<key> <comparator> <value>'." % identifier,
+            raise MlflowException("Invalid identifier '%s'. Columns should be specified as "
+                                  "'attribute.<key>', 'metric.<key>', 'tag.<key>', or "
+                                  "'param.'." % identifier,
                                   error_code=INVALID_PARAMETER_VALUE)
         identifier = cls._valid_entity_type(entity_type)
-        key = cls._strip_quotes(key)
+        key = cls._trim_backticks(cls._strip_quotes(key))
```

Comment:
key can support backticks too -- added a test for this.

Reply:
@mparkhe Didn't we agree that backticks are not supported: https://www.mlflow.org/docs/latest/search-syntax.html#entity-name-contains-special-characters

Reply:
Hmm, the reason to support backticks is because this is the SQL standard way for having column names with special characters - e.g. https://stackoverflow.com/questions/14190798/how-to-select-a-column-name-with-a-space-in-mysql